runtime/src/utils/buffer/write.rs

```diff
@@ -164,9 +149,27 @@ impl<B: Blob> Write<B> {
     /// This represents the total size of data that would be present after flushing.
     #[allow(clippy::len_without_is_empty)]
     pub async fn size(&self) -> u64 {
-        let inner = self.inner.read().await;
+        let buffer = self.buffer.read().await;
 
-        inner.size()
+        buffer.size()
+    }
+
+    /// Consumes the [Write] and returns the underlying blob.
+    ///
+    /// # Warning
+    ///
+    /// Any buffered data that hasn't been flushed or synced will be discarded.
+    pub fn take_blob(self) -> B {
+        self.blob
+    }
+
+    /// Clones the underlying blob.
+    ///
+    /// # Warning
+    ///
+    /// The returned blob will not include any data that has yet to be flushed from the buffer.
+    pub fn clone_blob(&self) -> B {
```

Comment:
If we implement `AsRef<Blob>`, someone can trivially clone without us having to do it for them.